demisto_sdk/commands/content_graph/interface/neo4j/queries/validations.py

```diff
@@ -124,6 +124,29 @@ def validate_toversion(
     }
 
 
+def get_items_used_deprecated(tx: Transaction, file_paths: List[str]):
+    files_filter = (
+        f"AND (p.path in {file_paths} OR d.path in {file_paths})" if file_paths else ""
+    )
+    command_query = f"""// Returning all the items which using deprecated commands
+    match (p{{deprecated: false}})-[:USES]->(c:Command)<-[:HAS_COMMAND{{deprecated: true}}]-(d:Integration) where not p.is_test
+optional match (i2:Integration)-[:HAS_COMMAND{{deprecated: false}}]->(c)
+with p, c, i2
+where i2 is null
+{files_filter}
+return c.object_id as deprecated_command, collect(p.path) as object_using_deprecated
+"""
+    general_items_query = f"""
+    match (p{{deprecated: false}})-[:USES]->(d{{deprecated: true}}) where not p.is_test
+optional match (p)-[:USES]->(c1:Command)<-[:HAS_COMMAND]-(d)
+with p, d, c1
+where c1 is null
+{files_filter}
+return d.path as deprecated_content, d.content_type as deprecated_content_type, collect(p.path) as object_using_deprecated, p.content_type as object_using_deprecated_type
```

Comment:
please do the uppercase keywords here as well.
why do we need this part: `optional match (p)-[:USES]->(c1:Command)<-[:HAS_COMMAND]-(d)`?
@yaakovpraisler

Reply:
This is to filter USES to integrations that their relationship with the playbook is because the playbook uses their command.

Since we create a USES relationship from the playbook to an integration as well as with the command (even if the only relationship is that the playbook calls the command) this result should be already calculated in the command's query, this is why we're filtering this result.